src/app/shared/services/base.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { environment } from '../../../environments/environment';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { catchError, retry } from 'rxjs/operators';
+import { map } from 'rxjs/operators';
+import { Deserializer } from 'ts-jsonapi';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export abstract class BaseService {
+  DEFAULT_HEADER: object = {
+    'Content-Type': 'application/json'
+  };
+
+  deserializer: Deserializer;
+
+  constructor(private http: HttpClient) {
+    this.deserializer = new Deserializer({
+      keyForAttribute: 'camelCase'
+    });
+  }
+
+  post(endpoint: string, body: {}): Observable<any> {
+    const apiUrl = this.apiUrlFor(endpoint);
+
+    return this.http.post(apiUrl, body).pipe(
```

Comment:
Referring to my above comment, what do you think of renaming `body` to `payload`? It's slightly more descriptive and less generic. It's a minor naming improvement but combined with the tests, I find it clearer.